internal/backends/sqlite/metadata/registry.go

```diff
@@ -542,6 +542,75 @@ func (r *Registry) indexesDrop(ctx context.Context, dbName, collectionName strin
 	return nil
 }
 
+// IndexesDropByNames remove given connection's indexes but their names.
+func (r *Registry) IndexesDropByNames(ctx context.Context, dbName, collectionName string, indexes []string) error {
```

Comment:
I think we can have one method that accepts `Indexes []IndexInfo` and won't need three methods `IndexesDropByNames`, `IndexesDropBySpec` and `IndexesDropAll`.

This is optional comment. It can be todo for a future to reduce complexity a bit.

Reply:
I think it should be a list of index names because they are unique for collection, and the caller (handler) can always call ListIndexes to get them

Reply:
Yep certainly cleaner!